compiler/rustc_codegen_ssa/src/mir/place.rs

```diff
@@ -471,3 +437,73 @@ fn round_up_const_value_to_alignment<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     let offset = bx.and(neg_value, align_minus_1);
     bx.add(value, offset)
 }
+
+/// Calculates the value that needs to be stored to mark the discriminant.
+///
+/// This might be `None` for a `struct` or a niched variant (like `Some(&3)`).
+///
+/// If it's `Some`, it returns the value to store and the field in which to
+/// store it. Note that this value is *not* the same as the discriminant, in
+/// general, as it might be a niche value or have a different size.
+///
+/// It might also be an `Err` because the variant is uninhabited.
+pub(super) fn codegen_tagged_field_value<'tcx, V>(
```

Comment:
mmh, `tagged_field_value` feels... off? I know it's an excruciatingly small detail, but `codegen_tag_value` sounds more correct... namely, like it doesn't codegen anything in the univariant case, which is important?